hubspot/serializers.py

```diff
@@ -50,6 +51,133 @@ class Meta:
         model = models.Line
 
 
+class B2BProductVersionToLineSerializer(serializers.ModelSerializer):
+    """ B2B product version to line serializer for Hubspot """
+
+    id = serializers.SerializerMethodField()
+    product = serializers.SerializerMethodField()
+    order = serializers.SerializerMethodField()
+    quantity = serializers.SerializerMethodField()
+    status = serializers.SerializerMethodField()
+    product_id = serializers.SerializerMethodField()
+
+    def get_id(self, instance):
+        """ Get the product version id"""
+        return format_hubspot_id(instance.product_version.id)
+
+    def get_product(self, instance):
+        """ Get the product id and return the hubspot product integratorObject id"""
+        return format_hubspot_id(instance.product_version.product.id)
+
+    def get_order(self, instance):
+        """ Get the order id and return the hubspot deal integratorObject id"""
+        return format_hubspot_id(instance.id)
+
+    def get_quantity(self, instance):
+        """return the number of seats associated with the b2b order"""
+        return instance.num_seats
+
+    def get_status(self, instance):
+        """ Get status of the associated Order """
+        return instance.status
+
+    def get_product_id(self, instance):
+        """Return the product version text_id"""
+        if instance.product_version:
+            return instance.product_version.text_id
+        return ""
+
+    class Meta:
+        fields = ("id", "product", "order", "quantity", "status", "product_id")
+        model = b2b_models.B2BOrder
+
+
+class B2BOrderToDealSerializer(serializers.ModelSerializer):
+    """ B2BOrder/Deal Serializer for Hubspot """
+
+    name = serializers.SerializerMethodField()
+    stage = serializers.SerializerMethodField()
+    close_date = serializers.SerializerMethodField(allow_null=True)
+    amount = serializers.SerializerMethodField()
+    discount_amount = serializers.SerializerMethodField()
+    discount_percent = serializers.SerializerMethodField()
+    coupon_code = serializers.SerializerMethodField(allow_null=True)
+    company = serializers.SerializerMethodField(allow_null=True)
+    payment_type = serializers.SerializerMethodField(allow_null=True)
+    payment_transaction = serializers.SerializerMethodField(allow_null=True)
```

Comment:
Add `purchaser` field similar to that in `OrderToDealSerializer` so that a contact is associated with the deal

Reply:
If the contact is not in our system, we may have only the email address in bulk purchases.

Reply:
@mbertrand addressed your feedback except this one, I am confused, we do not have purchaser id except the email field so would it be possible to link in hubspot ?

Reply:
Sorry, I was assuming that the email field would be for a user in the bootcamp database, but that's isn't necessarily the case.  I think you should ask @pdpinch what to do here - several options might be: a) not have a contact and leave as is or b) include the email as an additional field in the deal.

Reply:
Sorry, I don't think I understand the question. Ideally, we would want to create a contact in Hubsport for the bulk purchaser's email. Is that possible?

Reply:
The only info we have is email, and the contact details sent to hubspot includes a [unique id, based on the user id](https://github.com/mitodl/mitxpro/blob/da1bbd5ce1c7cfd6737c1fc2279fd1ba5dc36c1f/hubspot/api.py#L127):

```python
"integratorObjectId": format_hubspot_id(object_id)
```

Maybe it would be possible to use the email as the object_id?  Not sure what would happen if there's already a contact in hubspot with the same email address but a different object id.
This contact would also have no name, email would be the only populated field.  Might be worth a try.

Reply:
I think, It works over the ID

Reply:
The easiest solution might be to add the email as an additional field on the deal, but that seems a bit hacky.  Otherwise, to create a contact based on that email and associate it with the deal, I think a few things would need to be changed/added:

- Instead of this:
```python
        if not User.objects.filter(email=email).exists():
            send_user_to_hubspot(request=request, **{"email": email})
```
- do something like this:
```
        if not User.objects.filter(email=email).exists():
          hubspot.tasks.sync_b2b_contact_with_hubspot(email)
```
based loosely on `hubspot.tasks.sync_contact_with_hubspot(user_id)`, with a `make_b2b_contact_sync_message` function that uses the email (or maybe a b64 encoded string of the email) as the id.  Then use that same id for the "purchaser" value in the deal.

```python
def make_b2b_contact_sync_message(email):
   user_id = b64encode(email.encode("utf8"))
   return [make_sync_message(user_id, {"email": email})]
```